### ISSUE TITLE
click to expect the hover display information to be dynamically updated without refreshing the hover.

### ISSUE THREAD

**Issue by OceanBelongsToMe** [microsoft/vscode#203767]
As shown in the picture, click to expect the hover display information to be dynamically updated without refreshing the hover.

<img width="435" alt="image" src="https://github.com/microsoft/vscode/assets/22573382/e0403807-a784-45dc-b38d-fb9198deb7cb">

**Comment by aiday-mar** [comment]
Hi, thank you for writing this issue. Could you please include a bigger screenshot, and even preferably a video screen recording of the issue? I am not quite sure I understand where the issue is arising.

**Comment by OceanBelongsToMe** [comment]
When the hover content is updated, hover again and flashing occurs.

https://github.com/microsoft/vscode/assets/22573382/423b5a14-741a-4b9f-939a-4aa3ea5c3a5a

**Comment by OceanBelongsToMe** [comment]
Thank you very much for your reply😁

**Comment by aiday-mar** [comment]
I see thank you for the update

**Comment by OceanBelongsToMe** [comment]
It seems to be the same issue as #195394. After the official version is released, I will verify it.

**Comment by aiday-mar** [comment]
This API will have to be adopted by the extension authors for the bug to be fixed.

**Comment by OceanBelongsToMe** [comment]
Ok I'll try it in my own extension

**Comment by aiday-mar** [comment]
Yes when it comes out, the API has not come out yet. You can track the state of that work on the issue you linked.

**Comment by OceanBelongsToMe** [comment]
🫡，谢啦！

**Comment by OceanBelongsToMe** [comment]
Verbosity Hover 能够实现我的需求，但在 automatic completion 中就失效了，其实，editor.action.showHover 就能满足，但在 hover Widget 中，这个命令并不会重新渲染 hover

能否让hover中命令文本：editor.action.showHover，能触发重新渲染hover

**Comment by aiday-mar** [comment]
Hi @OceanBelongsToMe apologies I do not speak Mandarin. Can you please explain your comment in English?

**Comment by OceanBelongsToMe** [comment]
Verbosity Hover can meet my needs, but it fails in Completion Hover. In fact, editor.action.showHover can satisfy it, but in hover Widget, this command will not re-render hover Widget.

Can the command text in hover: editor.action.showHover trigger re-rendering of hover?

**Comment by aiday-mar** [comment]
Hi @OceanBelongsToMe I believe the current command already re-renders the hover when it is invoked.

In the example you gave in the screen recording above, how do you update the hover contents (what is the code you use to update the hover contents?)? 

A potential feature request would be to update the contents of the hover without hiding the hover before and creating a new one.

**Comment by OceanBelongsToMe** [comment]
For example, the following markdown text is displayed through hover. Clicking the command "editor.action.showHover" will send a hover request to lsp, but after receiving the new hover, it will not be rendered.
```
## 测试

- [editor.action.showHover](command:editor.action.showHover)
```

**Comment by OceanBelongsToMe** [comment]
> Hi @OceanBelongsToMe I believe the current command already re-renders the hover when it is invoked.
> 
> In the example you gave in the screen recording above, how do you update the hover contents (what is the code you use to update the hover contents?)?
> 
> A potential feature request would be to update the contents of the hover without hiding the hover before and creating a new one.

I have added a custom command in the hover: 'command:command:my.action.showNewHover?xxxxx'. This command carries some parameters and will trigger another hover event (sending a hover request to the LSP and updating the hover content).

However, the built-in event 'editor.action.showHover' does not re-render the hover after obtaining the new hover content.

**Comment by Helban** [comment]
+1 on this. I'm building an extension with interactive hover content (blame history navigation — clicking "← Previous" updates the underlying data). Currently there's no way to re-render the hover without the user moving the mouse away and back.

Worth noting that GitLens has the same unsolved problem. Their refreshHover command is literally:

typescript
// TODO@eamodio figure out how to really refresh/update a hover
await executeCoreCommand('editor.action.showHover');


Source: https://github.com/gitkraken/vscode-gitlens/blob/main/src/commands/refreshHover.ts

A new `editor.action.refreshHover` command (or an `onDidChangeHover` event on HoverProvider) would solve this for everyone.
